### PR TITLE
Added sum and product for degrade

### PR DESCRIPTION
**Pull request by fjaviersanchez**
I just added sum and product abilities to the `hsp.degrade` method.

### CODE REVIEW

File: tests/test_degrade.py [hunk 1]
```diff
@@ -286,6 +286,102 @@ def test_degrade_int_and(self):
...
+        sparse_map = healsparse.HealSparseMap(healpix_map=full_map, nside_coverage=nside_coverage,
+                                              nside_sparse=nside_map, sentinel=0)
+
+        # Degrade original HEALPix map
+
+        deg_map = np.full(hp.nside2npix(nside_new), 2**4, dtype=np.int64)
+        # Degrade sparse map and compare to original
+
+        new_map = sparse_map.degrade(nside_out=nside_new, reduction='prod')
+        print(new_map.generate_healpix_map())
```
Comment (erykoff) [inline tests/test_degrade.py]: I assume this is a debugging statement?  It should be removed.

File: tests/test_degrade.py [hunk 2]
```diff
@@ -286,6 +286,102 @@ def test_degrade_int_and(self):
...
+        sparse_map = healsparse.HealSparseMap(healpix_map=full_map, nside_coverage=nside_coverage,
+                                              nside_sparse=nside_map, sentinel=0)
+
+        # Degrade original HEALPix map
+
+        deg_map = np.full(hp.nside2npix(nside_new), 4, dtype=np.int64)
+        # Degrade sparse map and compare to original
+
+        new_map = sparse_map.degrade(nside_out=nside_new, reduction='sum')
+        print(new_map.generate_healpix_map())
```
Comment (erykoff) [inline tests/test_degrade.py]: Ditto.

**Review by erykoff — APPROVED** [review]
Looks good, but the debugging statements should be removed from the tests.

File: tests/test_degrade.py (hunk 2 above)
Comment (fjaviersanchez) [reply tests/test_degrade.py]: Thanks! Sorry, I forgot I included them!

**Pull request merged**